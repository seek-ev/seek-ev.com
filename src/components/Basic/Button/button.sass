// Variables
$opacity: 0.3

// Basic button styles
.btn
    border: 0px
    outline: none
    font-size: 16px
    color: #212121
    font-weight: 500
    border-radius: 7px
    margin-bottom: 12px
    padding: 7px 1.6rem
    letter-spacing: .5px
    background-color: #fff
    text-transform: uppercase
    transition: all 0.3s ease 0s
    font-family: 'Roboto', sans-serif
    box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07)
    -moz-box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07)
    -webkit-box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07)

.btn:hover
    color: #fff
    background-color: #1DB954
    box-shadow: 1px 2px 4px 2px rgba(0,0,0, 0.12)
    -moz-box-shadow: 1px 2px 4px 2px rgba(0,0,0, 0.12)
    -webkit-box-shadow: 1px 2px 4px 2px rgba(0,0,0, 0.12)

// Disabled
.btn:disabled
    color: #212121 !important
    background-color: #fff !important
    box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07) !important
    -moz-box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07) !important
    -webkit-box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07) !important

.btn:disabled:hover
    color: #212121
    background-color: #fff
    box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07)
    -moz-box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07)
    -webkit-box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07)

// Clean button
.clean-button:hover
    color: #212121 !important
    background-color: #fff !important
    box-shadow: 0px 2px 8px 1px rgba(0,0,0,0.15)
    -moz-box-shadow: 0px 2px 8px 1px rgba(0,0,0,0.15)
    -webkit-box-shadow: 0px 2px 8px 1px rgba(0,0,0,0.15)

// Button animations
.raise-button
    transition: all 0.3s ease 0s

.raise-button:hover
    transform: translateY(-6px)
    box-shadow: 1px 2px 4px 2px rgba(0,0,0, 0.12)
    -moz-box-shadow: 1px 2px 4px 2px rgba(0,0,0, 0.12)
    -webkit-box-shadow: 1px 2px 4px 2px rgba(0,0,0, 0.12)

.raise-button:disabled
    transition: none

.raise-button:disabled:hover
    transform: none

// Primary button
.primary-button
    color: #fff
    background-color: #1DB954

.primary-button:hover
    background-color: #1DB954e3
    box-shadow: 1px 2px 4px 2px rgba(30, 211, 93, $opacity)
    -moz-box-shadow: 1px 2px 4px 2px rgba(30, 211, 93, $opacity)
    -webkit-box-shadow: 1px 2px 4px 2px rgba(30, 211, 93, $opacity)

// Error button
.error-button
    color: #fff
    background-color: #ed1c46

.error-button:hover
    background-color: #ed1c46e3
    box-shadow: 0px 2px 8px 1px rgba(237, 28, 70, $opacity)
    -moz-box-shadow: 0px 2px 8px 1px rgba(237, 28, 70, $opacity)
    -webkit-box-shadow: 0px 2px 8px 1px rgba(237, 28, 70, $opacity)

// Dark button
.dark-button
    color: #fff
    background-color: #080808

.dark-button:hover
    background-color: #080808e3
    box-shadow: 0px 2px 8px 1px rgba(8, 8, 8, $opacity)
    -moz-box-shadow: 0px 2px 8px 1px rgba(8, 8, 8, $opacity)
    -webkit-box-shadow: 0px 2px 8px 1px rgba(8, 8, 8, $opacity)

// Blue button
.blue-button
    color: #fff
    background-color: #0070f3

.blue-button:hover
    background-color: #0071f3e3
    box-shadow: 0px 2px 8px 1px rgba(0, 113, 243, $opacity)
    -moz-box-shadow: 0px 2px 8px 1px rgba(0, 113, 243, $opacity)
    -webkit-box-shadow: 0px 2px 8px 1px rgba(0, 113, 243, $opacity)

// Round button
.round-button
    border-radius: 16px
