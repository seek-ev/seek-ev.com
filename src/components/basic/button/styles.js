import styled from 'styled-components/macro'

// Variables
import { colors } from 'assets/variables'

const opacity = 0.3

const Wrapper = styled.button`
    border: 0px;
    outline: none;
    font-size: 16px;
    color: #212121;
    font-weight: 500;
    border-radius: 7px;
    margin-bottom: 12px;
    padding: 7px 1.6rem;
    letter-spacing: .5px;
    background-color: #fff;
    text-transform: uppercase;
    transition: all 0.3s ease 0s;
    font-family: 'Roboto', sans-serif;
    box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07);
    -moz-box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07);
    -webkit-box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07);

    :hover {
        color: #fff;
        background-color: ${colors.primary};
        box-shadow: 1px 2px 4px 2px rgba(0,0,0, 0.12);
        -moz-box-shadow: 1px 2px 4px 2px rgba(0,0,0, 0.12);
        -webkit-box-shadow: 1px 2px 4px 2px rgba(0,0,0, 0.12);
    }

    :disabled {
        color: #212121 !important;
        background-color: #fff !important;
        box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07) !important;
        -moz-box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07) !important;
        -webkit-box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07) !important;
    }

    :disabled:hover {
        color: #212121;
        background-color: #fff;
        box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07);
        -moz-box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07);
        -webkit-box-shadow: 0px 1px 6px 2px rgba(0,0,0,0.07);
    }

    ${props => props.clean ? `
        :hover {
            color: #212121 !important;
            background-color: #fff !important;
            box-shadow: 0px 2px 8px 1px rgba(0,0,0,0.15);
            -moz-box-shadow: 0px 2px 8px 1px rgba(0,0,0,0.15);
            -webkit-box-shadow: 0px 2px 8px 1px rgba(0,0,0,0.15);
        }
    ` : ''}

    ${props => props.raise ? `
        transition: all 0.3s ease 0s;
        :hover {
            transform: translateY(-6px);
            box-shadow: 1px 2px 4px 2px rgba(0,0,0, 0.12);
            -moz-box-shadow: 1px 2px 4px 2px rgba(0,0,0, 0.12);
            -webkit-box-shadow: 1px 2px 4px 2px rgba(0,0,0, 0.12);
        }
    ` : ''}

    ${props => props.primary ? `
        color: #fff;
        background-color: ${colors.primary};

        :hover {
            background-color: #1DB954e3;
            box-shadow: 1px 2px 4px 2px rgba(30, 211, 93, ${opacity});
            -moz-box-shadow: 1px 2px 4px 2px rgba(30, 211, 93, ${opacity});
            -webkit-box-shadow: 1px 2px 4px 2px rgba(30, 211, 93, ${opacity});
        }
    ` : ''}

    ${props => props.error ? `
        color: #fff;
        background-color: ${colors.error};

        :hover {
            background-color: #ed1c46e3;
            box-shadow: 0px 2px 8px 1px rgba(237, 28, 70, ${opacity});
            -moz-box-shadow: 0px 2px 8px 1px rgba(237, 28, 70, ${opacity});
            -webkit-box-shadow: 0px 2px 8px 1px rgba(237, 28, 70, ${opacity});
        }
    ` : ''}

    ${props => props.dark ? `
        color: #fff;
        background-color: #080808;

        :hover {
            background-color: #080808e3;
            box-shadow: 0px 2px 8px 1px rgba(8, 8, 8, ${opacity});
            -moz-box-shadow: 0px 2px 8px 1px rgba(8, 8, 8, ${opacity});
            -webkit-box-shadow: 0px 2px 8px 1px rgba(8, 8, 8, ${opacity});
        }
    ` : ''}

    ${props => props.blue ? `
        color: #fff;
        background-color: #0070f3;

        :hover {
            background-color: #0071f3e3;
            box-shadow: 0px 2px 8px 1px rgba(0, 113, 243, ${opacity});
            -moz-box-shadow: 0px 2px 8px 1px rgba(0, 113, 243, ${opacity});
            -webkit-box-shadow: 0px 2px 8px 1px rgba(0, 113, 243, ${opacity});
        }
    ` : ''}

    ${props => props.round ? 'border-radius: 16px;' : ''}

    ${props => props.mr ? `
        margin-right: ${props.mr}px;
    ` : ''}

    ${props => props.ml ? `
        margin-left: ${props.ml}px;
    ` : ''}
`

export { Wrapper }