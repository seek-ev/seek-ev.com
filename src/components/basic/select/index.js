import React from 'react'

// Styles
import './select.sass'

const Select = (props) => {
  function handleChange(e) {
    // onChange prop is required
    props.onChange(e.target)
  }

  return (
    <select
      name={props.name}
      onChange={handleChange}
      disabled={props.disabled}
      className={props.raise ? 'raise-select ' : ''}
    >
      {props.options
        ? props.options.map((option, index) => (
          <option
            key={option.id ? option.id : index}
            hidden={option.hidden}
            value={
              props.value ? option[props.value] : option.value ? option.value : option.id ? option.id : option
            }
          >
            {props.display ? option[props.display] : option.name ? option.name : option}
          </option>
        ))
        : ''}
    </select>
  )
}

export { Select }
