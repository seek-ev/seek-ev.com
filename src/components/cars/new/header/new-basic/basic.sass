@import 'src/assets/variables'

.new-header-basic
    width: 100%
    display: flex
    flex-direction: column

.new-header-basic-row
    display: flex
    flex-direction: row
    @media only screen and (max-width: $max-desktop)
        flex-direction: column

.new-header-basic-input
    min-width: 260px
    margin: 0 64px 24px 0
    span
        margin: 0 0 3px
        font-weight: 500
    @media only screen and (max-width: $max-medium)
        margin: 0

.new-header-basic-select
    width: 260px
    margin: 0 64px 24px 0
    span
        margin: 0 0 3px
        font-weight: 500
    @media only screen and (max-width: $max-medium)
        width: 100%
        margin: 0 0 12px
