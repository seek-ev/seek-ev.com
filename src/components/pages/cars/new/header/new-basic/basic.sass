@import 'src/assets/variables'

.new-header-basic
    width: 100%
    display: flex
    flex-direction: column

.new-header-basic-row
    display: flex
    flex-direction: row
    justify-content: space-between
    @media only screen and (max-width: $max-medium)
        flex-direction: column

.new-header-basic-input
    min-width: 260px
    margin: 0 64px 24px 0
    span
        margin: 0 0 3px
        font-weight: 500
    @media only screen and (max-width: $max-medium)
        margin: 0

.new-header-basic-input-checkbox
    display: flex
    max-width: 250px
    span
        white-space: nowrap
        margin: 0 8px 3px 0
    input
        -ms-transform: scale(2)
        -moz-transform: scale(2)
        -webkit-transform: scale(2)
        -o-transform: scale(2)
        transform: scale(2)

.new-header-basic-select
    width: 260px
    margin: 0 64px 24px 0
    span
        margin: 0 0 3px
        font-weight: 500
    @media only screen and (max-width: $max-medium)
        width: 100%
        margin: 0 0 12px
