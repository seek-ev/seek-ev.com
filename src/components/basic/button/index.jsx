import React from 'react'

// Styles
import { Wrapper } from './styles'

const Button = (props) => {
  return (
    <Wrapper
      primary={props.primary}
      error={props.error}
      raise={props.raise}
      blue={props.blue}
      dark={props.dark}
      clean={props.clean}
      round={props.round}
      onClick={props.onClick}
      disabled={props.disabled}
    >
      {props.text}
    </Wrapper>
  )
}

export { Button }
