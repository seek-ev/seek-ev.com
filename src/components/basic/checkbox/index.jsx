import React from 'react'

// Styles
import { Wrapper, Title, CheckboxContainer, DefaultCheckbox, StyledCheckbox, Icon } from './styles'

const Checkbox = ({ className, direction, title, checked, name, onChange, disabled }) => {
    const change = (e) => {
        onChange(e.target)
    }

    return (
        <Wrapper className={className} direction={direction}>
            <Title>{title}</Title>
            <CheckboxContainer>
                <DefaultCheckbox checked={checked} name={name} onChange={change} disabled={disabled} />
                <StyledCheckbox checked={checked} disabled={disabled}>
                    <Icon viewBox="0 0 24 24">
                        <polyline points="20 6 9 17 4 12" />
                    </Icon>
                </StyledCheckbox>
            </CheckboxContainer>
        </Wrapper>
    )
}

export { Checkbox }