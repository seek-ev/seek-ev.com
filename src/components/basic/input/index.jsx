import React from 'react'

// Styles
import { Wrapper, Title, SInput, ErrorMessage } from './styles'

const Input = (props) => {
  function handleChange(e) {
    props.onChange(e.target)
  }

  return (
    <Wrapper className={props.className} title={props.title}>
      {props.title ? <Title>
        {props.title}
      </Title> : ''}
      <SInput
        id={props.id}
        value={props.value}
        onChange={handleChange}
        raise={props.raise}
        border={props.border}
        error={props.error}
        valid={props.valid}
        round={props.round}
        name={props.name}
        type={props.type}
        checked={props.checked}
        placeholder={props.placeholder}
        required={props.required}
        disabled={props.disabled}
        autoComplete={props.autoComplete}
        min={props.min}
        max={props.max}
        maxLength={props.maxlength}
        minLength={props.minlength}
        step={props.step}
      />

      <ErrorMessage>{props.error}</ErrorMessage>
    </Wrapper>
  )
}

export { Input }
