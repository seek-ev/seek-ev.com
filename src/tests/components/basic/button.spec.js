import React from 'react'
import { shallow } from 'enzyme'

// Component
import { Button } from 'components/basic/button'

describe('Button Component', () => {
    it('Render button correctly', () => {
        const button = shallow(<Button />)
        expect(button).toMatchSnapshot()
    })

    it('Displays text', () => {
        const button = shallow(<Button text="text" />)
        expect(button.text()).toEqual('text')
    })

    it('Click event', () => {
        const mockCallback = jest.fn()
        const button = shallow(<Button onClick={mockCallback} />)
        button.simulate('click')
        expect(mockCallback.mock.calls.length).toEqual(1)
    })

    it('Disabled Props', () => {
        const mockCallback = jest.fn()
        const button = shallow(<Button onClick={mockCallback} />)
        button.setProps({ disabled: true })
        expect(mockCallback.mock.calls.length).toEqual(0)
    })

    it('Primary props', () => {
        const button = shallow(<Button primary />)
        expect(button).toHaveStyleRule('background-color', '#1DB954')
    })

    it('Error props', () => {
        const button = shallow(<Button error />)
        expect(button).toHaveStyleRule('background-color', '#ed1c46')
    })

    it('Raise props', () => {
        const button = shallow(<Button raise />)
        expect(button).toMatchSnapshot()
    })

    it('Blue props', () => {
        const button = shallow(<Button blue />)
        expect(button).toHaveStyleRule('background-color', '#0070f3')
    })

    it('Dark props', () => {
        const button = shallow(<Button dark />)
        expect(button).toHaveStyleRule('background-color', '#080808')
    })

    it('Clean props', () => {
        const button = shallow(<Button clean />)
        expect(button).toMatchSnapshot()
    })

    it('Round props', () => {
        const button = shallow(<Button round />)
        expect(button).toHaveStyleRule('border-radius', '16px')
    })
})