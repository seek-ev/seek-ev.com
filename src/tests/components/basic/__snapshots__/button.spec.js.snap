// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Clean props 1`] = `ShallowWrapper {}`;

exports[`Button Component Raise props 1`] = `ShallowWrapper {}`;

exports[`Button Component Render button correctly 1`] = `ShallowWrapper {}`;
